### ISSUE TITLE
\ref command doesn't perform in LaTeX as described in the documentation

### ISSUE THREAD

**Issue by dhowland** [doxygen/doxygen#9726]
**Describe the bug**
The Doxygen manual states the following for the \ref command:

"For LaTeX documentation the reference command will generate a section number for sections or the text followed by a page number if <name> refers to an anchor."

This does not occur.  Section numbers are not generated.  Only the section title is used.

**Expected behavior**
In the example shown below, I would expect a `\ref sec13` command to produce "1.3 The Return of the King" in the LaTeX document.

**Screenshots**
For the following input:
```
/**

\page test1 Reference Test

\section sec11 The Fellowship of the Ring

words words words

To be continued in \ref sec12

\section sec12 The Two Towers

words words words

To be continued in \ref sec13

\section sec13 The Return of the King

words words words

*/
```
The following LaTeX is produced:
![image](https://user-images.githubusercontent.com/1534621/207972024-e0a9c4ec-5a33-41fc-b2b3-9d74b78ab75e.png)

**To Reproduce**
File attached.
[reftest.zip](https://github.com/doxygen/doxygen/files/10240810/reftest.zip)

**Version**
Doxygen 1.9.5

**Stack trace**
N/A

**Additional context**
N/A

**Comment by dhowland** [comment]
It occurred to me that this might be due to PDF_HYPERLINKS, so I turned that off and the result still doesn't produce a section number but rather a page number as described in the manual for anchors.

**Comment by dhowland** [comment]
So based on the `.tex` files that Doxygen creates, I looks like this is just a matter of putting the `\ref{label}` into the output.  However I see why this deviation from the manual was probably intentional.  I wonder would it make sense to make an option to choose either the title hyperlink or the number? Like `\ref['{'option'}'] <name> ["(text)"]` where option could be like "latexnum" or "secnum" or something like that.

In the mean time I supposed I could make a custom command along the lines of this:

`ALIASES += latexref{1}="\latexonly\ref\{\1\}\endlatexonly"`

**Comment by albert-github** [comment]
I think the base of the problem lies in the line:
> For LATEX documentation the reference command will generate a section number for sections or the text followed by a
page number if \<name\> refers to an anchor.

This line is already wrong as it doesn't mention that when `PDF_HYPERLINKS=YES` a hyperlink is generated (and no page number / section number at all). In case `PDF_HYPERLINKS=NO` a page number is shown.
I think the documentation should be updated:
- the part `section number` should be replaced by `section name` as this is how the current implementation is (I actually tried quite a few versions and never saw the `section number`)
- it should be mentioned that the page number is only shown in case of `PDF_HYPERLINKS=NO`.

@doxygen what do you think?
@dhowland is it for you absolutely necessary to have the section numbers as well?


(I didn't yet check RTF / docbook maybe we have similar problems here, but this could be solved in an analogous way)

**Comment by dhowland** [comment]
@albert-github I agree that in this case the documentation is probably incorrect, and the code is likely working as intended.

In my case it doesn't do what I need because I'm perhaps a slightly out-of-ordinary use case.  I have a significant portion of formal human-composed documentation that is attached to the API docs extracted with Doxygen.  So, I have big `.dox` files that just include chapters of technical writing.  Everything is published as informally as HTML and formally as PDF.  When used on the web, `/ref <section>` works perfectly.  As a large printable document with lots of pages in sequential sections, it's better to reference section number rather than section title.  ("see Section 4.2 for more information", etc.)  Of course that makes no sense on the web because the sections don't have numbers in HTML.

I solved the problem for my case with a collection of Aliases as I posted above.  I discovered how the LaTeX reference names are generated by Doxygen and thus created a `\ref_chap{<chapter_tag>}` command and `\ref_sec{<chapter_tag>,<section_tag>}` command.  This is not ideal but it works.   It doesn't really do exactly what I want (that is, as the documentation currently describes the `\ref` command) but I think with more hacking I might be able to get there.  Of course those solutions would be vulnerable to Doxygen updates changing how things work underneath.

So anyway, I don't expect you to change Doxygen to suit my needs, however it would be welcomed.  I can see needs for several kinds of behaviors, depending on the kind of documentation, which is why I suggested options as a possible solution.  Along those lines there it is also difficult to assign and reference LaTeX figure/table numbers, but I recognize that this stuff is probably outside of Doxygen's scope.

**Comment by albert-github** [comment]
Your use case is clear to me, though do you use `PDF_HYPERLINKS=YES` or `PDF_HYPERLINKS=NO`?

**Comment by dhowland** [comment]
I use `PDF_HYPERLINKS=YES` for two reasons.  One, they are useful when the PDF is viewed on PC, which is most of the time.  Two, I don't want page numbers in my text.

**Comment by albert-github** [comment]
The use of `PDF_HYPERLINKS=YES`  for sure makes sense especially when looking at a device that supports the possibility to jump between links.

I was looking into both versions. 
The version with `PDF_HYPERLINKS=NO` would be relatively easy for you to adjust as here the code looks like (see test1.tex):
```
\doxyref{The Two Towers}{p.}{test1_sec12}
```
and in the doxygen.sty we see the definition:
```
% Used when hyperlinks are turned off
\newcommand{\doxyref}[3]{%
  \textbf{#1} (\textnormal{#2}\,\pageref{#3})%
}
```

placing the following code in a new style file and adding this to the Doxyfile as `LATEX_EXTRA_STYLESHEET` would give the required output for you.
```
% Used when hyperlinks are turned off
\renewcommand{\doxyref}[3]{%
  \textbf{\ref{#3} #1}%
}
```

The solution for `PDF_HYPERLINKS=YES` is not that straightforward as here we have the code:
```
\mbox{\hyperlink{test1_sec12}{The Two Towers}}
```
and we would like to have
```
\mbox{\hyperlink{test1_sec12}{\ref{test1_sec12} The Two Towers}}
```
but as `\hyperlink` is a general command it is a bit hard to overrule, so we should create a doxygen specific  command here that could do the trick. 

Concluding:
- update documentation so that it reflects the current situation
- `PDF_HYPERLINKS=NO` is OK as it is and it is up to the doxygen user to change the `\DoxyRef`
- `PDF_HYPERLINKS=YES` should be adjusted in such a way that the doxygen user can adjust the hyperlink to his own preferences.

**Comment by albert-github** [comment]
At first glance the idea mentioned above looks OK, but it are not only section references that use the above mentioned constructs but also other types of anchors (like the `\anchor` command) and in that case we wouldn't like to have the section number but with the given workaround it does appear.
So some more thinking about the problem is required when we want to solve the issue inside doxygen.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9761

In your specific case you could place the following code in a new style file and adding this to the Doxyfile as `LATEX_EXTRA_STYLESHEET`: 
```
\renewcommand{\doxysectlink}[3]{%
  \ifthenelse{#3=1}%
  {%
  \mbox{\hyperlink{#1}{\ref{#1} #2}}%
  }%
  {%
  \mbox{\hyperlink{#1}{#2}}%
  }%
}
```

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by dhowland** [comment]
Sorry for the delay I was consumed with family things over the holiday.

I think I am following the changes so this should work.  The end result is similar to what I'm doing now with my aliased commands but definitely cleaner.  Thanks for the support.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.7.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
